### PR TITLE
[CERT] subscriber count functionality

### PR DESCRIPTION
**Pull request by MortenMunk**
## Description

Added subscriber count functionality + tests

## Changes

- removed one test that only tested response, and not actual database object
- consolidated courseRoutes.test.js and courseRoutes.spec.js to one file
- added tests for unsubscribing, and subscribing that tests both numOfSubscriptions for courses, and subscriptions for users
- added test for unsubscribing to a test without any subscribers
- added numOfSubscriptions: 0 as fakeCourse object did not specify this amount
- Added two error codes: -  Attempting to unsubscribe from a course you are not subscribed to - Attempting to subscribe to a course you are already subscribed to
- Updated subscribe routes to send the updated user instead of the old one
- Added an update to the coursemodels with an increment or decrement of one to the subscribe count for subscribe and unsubscribe respectively
- fixed indentation in some files

## Related Issues

None.

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

None.

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

Commit id: feb78487a83a3485ce6d3e8700f74318e9b64228

## Notes for Reviewers

None relevant.

### CODE REVIEW

File: __tests__/fixtures/fakeCourse.js [hunk 1]
```diff
@@ -8,9 +8,9 @@ module.exports = function makeFakeCourse() {
 		category: 'sewing',
 		published: false,
 		sections: [],
-		creator: [],
 		difficulty: 1,
-		time: 1,
+		estimatedHours: 1,
 		rating: 5,
+    numOfSubscriptions: 0,
```
Comment (PrometheusBork) [inline __tests__/fixtures/fakeCourse.js:14]: Very minor comment, "numOfSubsricptions" does not seem to be formatted correcly

**Review by PrometheusBork — APPROVED** [review]
Overall looks fine to me! Made a minor comment about formatting at a single line in fakeCourse.js, but this should not stop approval

File: __tests__/fixtures/fakeCourse.js (hunk 1 above)
Comment (MortenMunk) [reply __tests__/fixtures/fakeCourse.js:14]: No problem, I fixed it anyway 😎👍

**Review by JackSMunn — APPROVED** [review]
Looks good to me, didn't we fix the collapsing the 2 test files before this?

**Comment by MortenMunk** [comment]
> Looks good to me, didn't we fix the collapsing the 2 test files before this?

@sofiiagran mentioned she has merged them previously, but tests has been refactored to fit updated code, and several tests has been added, so I think it was in need of a merge anyway. This version is also more concise as it uses beforeAll, etc.

**Review by carlryskov — APPROVED** [review]

**Pull request merged**